File: serial-read.cpp
#include "pxt.h"

using namespace pxt;

namespace serial {

//%
int read() {
    int res = uBit.serial.read(MicroBitSerialMode::ASYNC);
    if (res >= 0) {
        return res;
    } else {
        return -1;
    }
}

//%
int available() {
    return uBit.serial.rxBufferedSize();
}

} // namespace serial

File: serial-read.ts
namespace serial {

  /**
  * Returns the first byte from the RX buffer or -1 if no data is available.
  */
  //% shim=serial::read
  export function read(): number  { return -1 }


  /**
  * Returns the number of currently available bytes in the RX buffer.
  */
  //% shim=serial::available
  export function available(): number { return 0 }
}